app/split_integer.py

```diff
@@ -1,7 +1,12 @@
 def split_integer(value: int, number_of_parts: int) -> list:
-    parts = []
-    for parts_left in range(number_of_parts, 0, -1):
-        next_number = value // parts_left
-        parts.append(value // parts_left)
-        value -= next_number
-    return parts
+    if value < number_of_parts:
+        return [0] * (number_of_parts - 1) + [value]
+
+    base_value = value // number_of_parts
+    remainder = value % number_of_parts
+
+    parts = [base_value] * number_of_parts
+    for i in range(remainder):
+        parts[i] += 1
+
+    return sorted(parts)
```

Comment:
The `sorted(parts)` function call will sort the parts in ascending order before returning them. If the task does not explicitly require the parts to be sorted, this operation is unnecessary and could be removed to improve performance.